### PR TITLE
shattered leagues fixes/additions

### LINKED ISSUE #14753: Cannon Plugin Broken - Cannon Shattered Ornament Kit
With the new Shattered Relics Cannon ornament kit release the existing cannon runelite plugin no longer works. Ammo is not tracked properly nor is double hit spots
image

### PR DESCRIPTION
**Pull request by Hydrox6**
Closes #14753
Closes #14759

Adds all the ornament kits from the league to Item Mapping (there's so many :d )
Fixed the cannon plugin to work with the ornamental cannon, which was more annoying than expected

### CODE REVIEW

**Comment by DavidLuong98** [comment]
ah I just pushed some changes for the ornamental cannon. seems like you got to it before me

**Review by Nightfirecat — COMMENTED** [review]
Haven't tested ingame, but lgtm

**Comment by Hydrox6** [comment]
> ah I just pushed some changes for the ornamental cannon. seems like you got to it before me

@DavidLuong98 Yeah, we needed it a little earlier than expected. It's also a lot more involved than you'd think; have a look at the extra changes I had to do to the CannonPlugin compared to your PR and you'll see why.

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java [hunk 1]
```diff
@@ -185,15 +191,19 @@ public void onItemContainerChanged(ItemContainerChanged event)
 				switch (item.getId())
 				{
 					case ItemID.CANNON_BASE:
+					case ItemID.CANNON_BASE_OR:
 						hasBase = true;
```
[the remark this answers is not in the document]
Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java:195]: Given that you can't put a cannon together with mismatched parts, and the spots tend to be in places away from banks where players would want to actually use the cannon, I doubt it'll actually come up very much. Even if it does happen, it's only the cannon spots overlay, which is incredibly minor to the user.

**Pull request merged**